src/rust-crypto/backup.ts

```diff
@@ -280,6 +308,93 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
             }
         }
     }
+
+    /**
+     * Creates a new key backup by generating a new random private key.
+     * If there is an existing backup server side it will be deleted and replaced
+     * by the new one.
+     *
+     * @param signer - Method that should sign the backup with existing device and
+     * existing identity.
+     * @returns a KeyBackupCreationInfo - All information related to the backup.
+     */
+    public async setUpKeyBackup(signer: (authData: AuthData) => Promise<void>): Promise<KeyBackupCreationInfo> {
+        // check first if there is an existing backup
+        if (this.activeBackupVersion || (await this.requestKeyBackupVersion()) != null) {
```

Comment:
this condition seems to be redundant? `deleteKeyBackup` will do it again, so it's just an extra HTTP hit for no reason?